diff --git a/src/Task.cpp b/src/Task.cpp
index 063ae1193..e4d631cc6 100644
--- a/src/Task.cpp
+++ b/src/Task.cpp
@@ -1880,7 +1880,7 @@ void Task::validate (bool applyDefault /* = true */)
     throw std::string ("Cannot add a task that is blank.");
 
   // Cannot have a recur frequency with no due date - when would it recur?
-  if (has ("recur") && (! has ("due") || get ("due") == ""))
+  if (has ("recur") && (! has ("due") || get ("due") == "") && (! has ("imask") || get ("imask") == ""))
     throw std::string ("A recurring task must also have a 'due' date.");
 
   // Recur durations must be valid.
diff --git a/src/commands/CmdModify.cpp b/src/commands/CmdModify.cpp
index cb407367d..7f2efd0d7 100644
--- a/src/commands/CmdModify.cpp
+++ b/src/commands/CmdModify.cpp
@@ -121,12 +121,6 @@ void CmdModify::checkConsistency (Task &before, Task &after)
       ! before.has ("due"))
     throw std::string ("You cannot specify a recurring task without a due date.");
 
-  if (before.has ("recur") &&
-      before.has ("due")   &&
-      (! after.has ("due")  ||
-        after.get ("due") == ""))
-    throw std::string ("You cannot remove the due date from a recurring task.");
-
   if (before.has ("recur")  &&
       (! after.has ("recur") ||
         after.get ("recur") == ""))
