diff --git a/src/openvpn/init.c b/src/openvpn/init.c
index 103aa3c..26affc8 100644
--- a/src/openvpn/init.c
+++ b/src/openvpn/init.c
@@ -767,7 +767,7 @@ context_init_1(struct context *c)
      * do any fork due to daemon() a future call.
      * See possibly_become_daemon() [init.c] for more details.
      */
-    sd_notifyf(0, "READY=1\nSTATUS=Pre-connection initialization successful\nMAINPID=%lu",
+    sd_notifyf(0, "STATUS=Pre-connection initialization successful\nMAINPID=%lu",
                (unsigned long) getpid());
 #endif
 
@@ -1618,6 +1618,9 @@ initialization_sequence_completed(struct context *c, const unsigned int flags)
     /* Flag that we initialized */
     if ((flags & (ISC_ERRORS|ISC_SERVER)) == 0)
     {
+#ifdef ENABLE_SYSTEMD
+        sd_notifyf(0, "READY=1");
+#endif
         c->options.no_advance = true;
     }
 
